repositories/iceoryx.BUILD.bazel

```diff
@@ -0,0 +1,51 @@
+load("@rules_cc//cc:defs.bzl", "cc_binary")
+load("@rules_foreign_cc//foreign_cc:cmake.bzl", "cmake")
+
+filegroup(
+    name = "all_srcs",
+    srcs = glob(["**"]),
+    visibility = ["//visibility:public"],
+)
+
+cmake(
+    name = "iceoryx",
+    build_args = [
+        "--",  # <- pass options to the native tool
+        "-j4",
+    ],
+    cache_entries = {
+        "BUILD_SHARED_LIBS": "OFF",
+        "CCACHE": "OFF",
+        "CMAKE_BUILD_TYPE": "Release",
+        "CMAKE_CXX_FLAGS": "-I$EXT_BUILD_DEPS/acl/include -L$EXT_BUILD_DEPS/acl/lib",
+    },
+    lib_source = ":all_srcs",
+    out_static_libs = [
+        "libiceoryx_binding_c.a",
+        "libiceoryx_posh_config.a",
+        "libiceoryx_posh_roudi.a",
+        "libiceoryx_posh.a",
+        "libiceoryx_posh_gateway.a",
+        "libiceoryx_hoofs.a",
+        "libiceoryx_platform.a",
```

Comment:
Please alph-sort the libs.

Reply:
Run into issue with this actually.

The order of `out_static_libs` in the cmake rule is the order being passed to the linker in the final cc_binary rule and this order matters. The rule is the library that needs symbols must be first, then the library that resolves the symbol after. Hence this list can only be in dependency order.